drivers/sensor/adxl372/adxl372.c

```diff
@@ -0,0 +1,1104 @@
+ * @param mode - settle period
+ *		Accepted values: ADXL372_FILTER_SETTLE_370
+ *				 ADXL372_FILTER_SETTLE_16
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int adxl372_set_filter_settle(struct device *dev,
+				  enum adxl372_filter_settle mode)
+{
+	return adxl372_reg_write_mask(dev, ADXL372_POWER_CTL,
+				      ADXL372_POWER_CTL_FIL_SETTLE_MSK,
+				      ADXL372_POWER_CTL_FIL_SETTLE_MODE(mode));
+}
+
+/**
+ * Configure the INT1 and INT2 interrupt pins.
+ * @param dev - The device structure.
+ * @param int1 -  INT1 interrupt pins.
+ * @param int2 -  INT2 interrupt pins.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int adxl372_interrupt_config(struct device *dev,
+				 u8_t int1,
+				 u8_t int2)
+{
+	int ret;
+
+	ret = adxl372_reg_write(dev, ADXL372_INT1_MAP, int1);
+	if (ret) {
+		return ret;
+	}
+
+	return  adxl372_reg_write(dev, ADXL372_INT2_MAP, int2);
+
+}
+
+/**
+ * Get the STATUS, STATUS2, FIFO_ENTRIES and FIFO_ENTRIES2 registers data
+ * @param dev - The device structure.
+ * @param status1 - Data stored in the STATUS1 register
+ * @param status2 - Data stored in the STATUS2 register
+ * @param fifo_entries - Number of valid data samples present in the
+ *			 FIFO buffer (0 to 512)
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int adxl372_get_status(struct device *dev,
+			   u8_t *status1,
+			   u8_t *status2,
+			   u16_t *fifo_entries)
+{
+	u8_t buf[4], length = 1;
+	int ret;
+
+	if (status2) {
+		length++;
+	}
+
+	if (fifo_entries) {
+		length += 2;
+	}
+
+	ret = adxl372_reg_read_multiple(dev, ADXL372_STATUS_1, buf, length);
+
+	*status1 = buf[0];
+
+	if (status2) {
+		*status2 = buf[1];
+	}
+
+	if (fifo_entries) {
+		*fifo_entries = ((buf[2] & 0x3) << 8) | buf[3];
+	}
+
+	return ret;
+}
+
+/**
+ * Software reset.
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int adxl372_reset(struct device *dev)
+{
+	int ret;
+
+	ret = adxl372_set_op_mode(dev, ADXL372_STANDBY);
+	if (ret) {
+		return ret;
+	}
+	/* Writing code 0x52 resets the device */
+	ret = adxl372_reg_write(dev, ADXL372_RESET, ADXL372_RESET_CODE);
+	k_sleep(1000);
+
+	return ret;
+}
+
+/**
+ * Get the data stored in FIFO.
+ * @param dev - The device structure.
+ * @param samples - pointer to the raw data stored in the ADXL372_FIFO_DATA
+ * @param cnt - How many samples should be retrieved from the FIFO DATA reg
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int adxl372_get_fifo_xyz_data(struct device *dev,
+			      struct adxl372_xyz_accel_data *samples,
+			      u16_t cnt)
+{
+	u8_t buf[128];
+	u16_t i, total;
+	int ret;
+
+	if (cnt > 512 || cnt == 0) {
+		return -EINVAL;
+	}
+	/*
+	 * The FIFO can hold up to 512 samples.
+	 * Each sample is 2 bytes, that's why we read (cnt * 2) bytes
+	 */
+
+	total = cnt * 2;
+
+	do {
+		if (total > sizeof(buf)) {
+			cnt = sizeof(buf);
```

Comment:
`cnt` appears to have a new meaning here, different from the function argument. Please use a new variable.

Reply:
ok